src/blockchain/Block.java

```diff
@@ -0,0 +1,56 @@
+package blockchain;
+
+import java.security.MessageDigest;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.List;
+import java.util.Random;
+
+public class Block {
+    private static final Random rand = new Random();
+    public final int minerId;
+    public final int id;
+    public final long timestamp;
```

Comment:
Is it possible to use LocalDateTime or Instant instead of long for current timestamp?

Reply:
Error with both
Operator '+' cannot be applied to 'int', 'java.time.LocalDateTime'
Operator '+' cannot be applied to 'int', 'java.time.Instant'

Reply:
Well of course there will be an error, because both of them are classes, not primitives. It's required to change some logic around this field